core/market/decentralized/src/matcher/error.rs

```diff
@@ -56,6 +62,14 @@ pub enum MatcherInitError {
     DatabaseError(#[from] DbError),
 }
 
+#[derive(thiserror::Error, Debug)]
+pub enum ResolverError {
+    #[error(transparent)]
+    OfferError(#[from] OfferError),
```

Comment:
OfferError has ResolverError and ResolverError has OfferError

Reply:
This shouldn't be possible. `MatcherError` has `ResolveError` which has `OfferError`

Reply:
I've tested and such setup results in compile error:
```
error[E0391]: cycle detected when processing `matcher::error::OfferError`
```